tests/endtoend/test_endtoend.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+import os
+from threading import Thread
+import requests
+import time
+import trello_items as trello
+from app import create_app
+from flask import current_app as app
+from dotenv import load_dotenv, find_dotenv
+from selenium import webdriver
+
+@pytest.fixture(scope='module')
+def test_app():
+    # Create the new board & update the board id environment variable
+    file_path = find_dotenv('.env')
+    load_dotenv(file_path, override=True)
+    
+    board_id = create_trello_board('My E2E Test Board')
+    update_env_vars(board_id)
+
+    # construct the new application
+    application = create_app()
+    
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False)) 
+    thread.daemon = True
+    thread.start()
+    yield app
+    # Tear Down
+    thread.join(1) 
+    delete_trello_board(board_id)
+
+@pytest.fixture(scope='module')
+def driver():
+    with webdriver.Firefox() as driver:
+        yield driver
+
+def update_env_vars(board_id):
+    os.environ['boardId'] = board_id
+    lists = trello.get_all_lists_on_board(board_id)
+    for list in lists:
+        if list['name'] == 'To Do':
+            os.environ['TODO_LIST_ID'] = list['id']
+        elif list['name'] == 'Doing':
+            os.environ['DOING_LIST_ID'] = list['id']
+        else:
+            os.environ['DONE_LIST_ID'] = list['id']
+
+def test_task_journey(driver, test_app): 
+    driver.get('http://127.0.0.1:5000/')
+    assert driver.title == 'To-Do App'
+
+    # Create new item
+    els = driver.find_elements_by_tag_name("td")
+    driver.find_element_by_id("name_input").send_keys("Watch movie")
+    driver.find_element_by_id("desc_input").send_keys("Movie on TV")
+    driver.find_element_by_id("add-item").click()
+    time.sleep(2)
+    els = driver.find_elements_by_tag_name("td")
+    assert driver.find_element_by_xpath("//td[2]").text == "Movie on TV"
```

Comment:
Would be slightly nicer to find the element inside todo-tbody specifically rather than just asserting that the todo item is inside any td[2] on the page
Same for the other asserts

Reply:
Done